File: CS/WinSolution.Module/DomainObjectBase.cs
using DevExpress.Xpo;
using DevExpress.Persistent.BaseImpl;

namespace WinSolution.Module {
    public abstract class DomainObjectBase : BaseObject {
        public DomainObjectBase(Session session) : base(session) { }
        [Association("DomainObjectBase-Files"), Aggregated]
        public XPCollection<File> Files {
            get {
                return GetCollection<File>("Files");
            }
        }
    }
}

File: VB/WinSolution.Module/DomainObjectBase.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Persistent.BaseImpl

Namespace WinSolution.Module
	Public MustInherit Class DomainObjectBase
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		<Association("DomainObjectBase-Files"), Aggregated> _
		Public ReadOnly Property Files() As XPCollection(Of File)
			Get
				Return GetCollection(Of File)("Files")
			End Get
		End Property
	End Class
End Namespace
